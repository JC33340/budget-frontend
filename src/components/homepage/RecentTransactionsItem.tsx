import type { transactionLogsType } from '../../pages/Transactions';

type RecentTransactionsItemType = {
    transactionLog: transactionLogsType;
    color: string;
};

const RecentTransactionsItem = ({
    transactionLog,
    color,
}: RecentTransactionsItemType) => {
    const date = new Date(transactionLog.created_at).toLocaleDateString(
        'en-GB',
        {
            month: 'short',
            day: 'numeric',
            year: 'numeric',
        }
    );
    return (
        <div className={`${color} grid grid-cols-3 rounded-lg`}>
            <p className="flex items-center p-2 border-r-2 border-lighter-gray">
                {transactionLog.value}
            </p>
            <p className="flex items-center capitalize p-2 border-r-2 border-lighter-gray">
                {transactionLog.tag}
            </p>
            <p className="flex items-center p-2">{date}</p>
        </div>
    );
};

export default RecentTransactionsItem;
