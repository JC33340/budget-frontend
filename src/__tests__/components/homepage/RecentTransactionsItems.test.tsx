import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import RecentTransactionsItem from '../../../components/homepage/RecentTransactionsItem';

const testObj = {
    value: 100,
    tag: 'test tag',
    created_at: new Date('16 janurary 2025').toLocaleDateString(),
    notes: 'test notes',
    id: 0,
};

test('snapshot test', () => {
    render(<RecentTransactionsItem color="" transactionLog={testObj} />);
    const item = screen.getByTestId('recentTransactionsItem-testid');
    expect(item).toMatchSnapshot();
});

test('test content', () => {
    render(<RecentTransactionsItem color="" transactionLog={testObj} />);
    const comp = screen.getByTestId('recentTransactionsItem-testid');
    expect(comp).toHaveTextContent('test tag');
    expect(comp).toHaveTextContent('100');
    expect(comp).toHaveTextContent('16 Jan 2025');
});
