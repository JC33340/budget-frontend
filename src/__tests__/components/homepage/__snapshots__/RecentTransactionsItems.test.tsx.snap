// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class=" grid grid-cols-3 rounded-lg"
  data-testid="recentTransactionsItem-testid"
>
  <p
    class="flex items-center p-2 border-r-2 border-lighter-gray"
  >
    100
  </p>
  <p
    class="flex items-center capitalize p-2 border-r-2 border-lighter-gray"
  >
    test tag
  </p>
  <p
    class="flex items-center p-2"
  >
    16 Jan 2025
  </p>
</div>
`;
